### PR TITLE
Add multiplayer turn sound notification

### PR DESCRIPTION
**Pull request by Azzurite**
I swear this is not as bad as the changed files/lines looks :D The main commit is only +279 -97 lines, the rest are refactorings.

This adds a notification sound for when it's your turn in one of your multiplayer games. This supplements the visual feedback that I added with the multiplayer status display, and is useful for people that have the game running on desktop in the background, or are leaving their phone lying on a table with screen turned on (both of which I like to do).

I again recommend strongly to review this commit-by-commit:

* [4d44be1](https://github.com/yairm210/Unciv/pull/6995/commits/4d44be118e47bd2c4f7b59c07d223162a58bf85e): A simple refactoring that eliminates the ugly `fun updateRefreshSelectOptions`
* [fee8e06](https://github.com/yairm210/Unciv/pull/6995/commits/fee8e068c749aa6a76c895e6fb05e44630aba182): This is in preparation for me adding a different `Sounds` class for handling game sounds
* [eaa5dcd](https://github.com/yairm210/Unciv/pull/6995/commits/eaa5dcdfa17db50ac97d921e2785dc29b0dcfb99): Removes the IMO unnecessary `UncivSoundConstants` and makes `UncivSound` simply fileName-based. To allow later serialization via JSON.
* [3bae2cb](https://github.com/yairm210/Unciv/pull/6995/commits/3bae2cbcb1edc365bb6df2d0e3fec16ef0f3b663): Reorder sounds by ABC
* [8a7f64c](https://github.com/yairm210/Unciv/pull/6995/commits/8a7f64c55e3e039bae516a85a51314ba9d85e42d): This is where all the changes come from, I split up `ExtensionFunctions` and moved it into a new package. Naturally, almost everything has to get its references updated. But nothing complex that needs review beyond maybe the actual splitting up.
* [e8d7d09](https://github.com/yairm210/Unciv/pull/6995/commits/e8d7d09c740e755e467b8bcb45007e933362d045): Actually adds the sound notification. I'll add some more code comments for this.

### CODE REVIEW

**Comment by yairm210** [comment]
Conflicts

File: core/src/com/unciv/ui/options/OptionsPopup.kt [hunk 1]
```diff
@@ -148,3 +160,71 @@ class OptionsPopup(
...
+        when (item) {
+            is UncivSound -> EventBus.send(object : SettingsPropertyUncivSoundChanged {
+                override val gameSetting = setting
+                override val value: UncivSound = settingsProperty.get() as UncivSound
+            })
+            else -> EventBus.send(object : SettingsPropertyChanged {
+                override val gameSetting = setting
+            })
+        }
+    }
```
[the remark this answers is not in the document]
Comment (yairm210) [reply core/src/com/unciv/ui/options/OptionsPopup.kt:229]: Sounds to me like if you're sending an event anyway, then like `val event = when (item){...} eventbus.send(event)`

File: core/src/com/unciv/UncivGame.kt [hunk 2]
```diff
@@ -274,6 +278,14 @@ class UncivGame(parameters: UncivGameParameters) : Game() {
...
         lateinit var Current: UncivGame
         fun isCurrentInitialized() = this::Current.isInitialized
+        fun isCurrentGame(gameId: String): Boolean {
+            return if (isCurrentInitialized() && Current.isGameInfoInitialized()) {
+                Current.gameInfo.gameId == gameId
+            } else {
+                false
+            }
+
+        }
```
Comment (yairm210) [inline core/src/com/unciv/UncivGame.kt:289]: Sounds to me like 'return isCurrentInitialized() && Current.isGameInfoInitialized() && Current.gameInfo.gameId == gameId`

Comment (yairm210) [reply core/src/com/unciv/UncivGame.kt:289]: Whenever you have an if/else where one of the options is just a boolean you know it can be simplified

Comment (SomeTroglodyte) [reply core/src/com/unciv/UncivGame.kt]: > need to get rid of the `lateinit` modifier

Oh, yes, wherever possible or impossible. And the God Object:tm: itself too! (Actually, it's not so bad)

**Comment by yairm210** [comment]
Conflicts

**Pull request merged**